url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwin/fitness/Fitness.java

```diff
@@ -0,0 +1,292 @@
+		return multiobjectiveStandard[population].get(individual);
+	}
+	
+	/**
+	 * Return an adjusted fitness value
+	 * @param metric	 the fitness metric index to access
+	 * @param population the population index to access
+	 * @param individual the individual index to access
+	 * @return			 the adjusted fitness, calculated from dominations
+	 */
+	public double getAdjustedFitness(int metric, int population, int individual) {
+		return adjustedFitness[metric][population].get(individual);
+	}
+	
+	/**
+	 * Return a adjusted multiobjective fitness value
+	 * @param population the population index to access
+	 * @param individual the individual index to access
+	 * @return			 the current raw fitness value
+	 */
+	public double getAdjustedFitness(int population, int individual) {
+		return multiobjectiveAdjusted[population].get(individual);
+	}
+	
+	/**
+	 * Return a normalized fitness value
+	 * @param metric	 the fitness metric index to access
+	 * @param population the population index to access
+	 * @param individual the individual index to access
+	 * @return			 the normalized fitness, calculated from dominations
+	 */
+	public double getNormalizedFitness(int metric, int population, int individual) {
+		return normalizedFitness[metric][population].get(individual);
+	}
+	
+	/**
+	 * Return a normalized multiobjective fitness value
+	 * @param population the population index to access
+	 * @param individual the individual index to access
+	 * @return			 the current raw fitness value
+	 */
+	public double getNormalizedFitness(int population, int individual) {
+		return multiobjectiveNormalized[population].get(individual);
+	}
+	
+	/**
+	 * Return complexity of an individual
+	 * @param population the population index to access
+	 * @param individual the individual index to access
+	 * @return			 the number of nodes this individual is made of
+	 */
+	public int getComplexity(int population, int individual) {
+		return complexity[population].get(individual);
+	}
+	
+	/**
+	 * Return the indexes of the best-of-generation individuals
+	 * @param population the population index to access
+	 * @param number	 the number of indexes to return
+	 * @return			 the indexes of the best individual
+	 */
+	public Collection<Integer> getBestOfGeneration(int population, int number) {
+		TreeMap<Double, Integer> best = new TreeMap<Double, Integer>();
+		if(number != 0) {
+			best.put(multiobjectiveNormalized[population].get(0), 0);
+			for(int i = 1; i < multiobjectiveNormalized[population].size(); i++) {
+				double curr = multiobjectiveNormalized[population].get(i);
+				if(best.size() < number || curr > best.firstKey())
+					best.put(curr, i);
+				if(best.size() > number)
+					best.remove(best.firstKey());
+			}
+		}
+		return best.values();
+	}
+	
+	/**
+	 * Distributes and processes all fitness-related calculations
+	 * @throws BusyException
+	 */
+	public void assessFitness() throws BusyException {	
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 215-227, lines 228-254, lines 257-268, lines 269-290 into separate methods.